bolts/callbacks/posthoc_eval.py

```diff
@@ -1,18 +1,78 @@
 from collections import Sequence
+from typing import NamedTuple, Optional
 
 from kit import gcopy
 import pytorch_lightning as pl
-from pytorch_lightning.utilities.types import STEP_OUTPUT
+from pytorch_lightning.utilities.types import EPOCH_OUTPUT, STEP_OUTPUT
+import torch
+from torch import Tensor, nn
+from torch.optim import AdamW
 from torch.utils.data import DataLoader
+import torchmetrics
 
 __all__ = ["PostHocEval"]
 
 
+class EvalTuple(NamedTuple):
```

Comment:
Use the newly-added BinaryTuple? This will require a merge with master.

Reply:
Where is BinaryTuple?

Reply:
BinarySample